addons/l10n_it_edi/models/account_invoice.py

```diff
@@ -263,6 +267,14 @@ def format_alphanumeric(text_to_convert):
         invoice_lines = self._l10n_it_edi_prepare_fatturapa_line_details(reverse_charge_refund, is_downpayment, convert_to_euros)
         tax_details = self._l10n_it_edi_prepare_fatturapa_tax_details(tax_details, reverse_charge_refund)
 
+        # If there was a down payed amount that's been deducted from this move,
+        # we need to put a reference to the down payment invoice in the DatiFattureCollegate tag
+        downpayment_moves = self.env['account.move']
+        if not is_downpayment:
+            negative_invoice_lines = self.invoice_line_ids.filtered(lambda l: l.price_subtotal < 0)
+            if negative_invoice_lines:
+                downpayment_moves = negative_invoice_lines.mapped(lambda l: l._get_downpayment_lines()).mapped("move_id")
```

Comment:
why not handle this as well in the changes above? Because here you need it for all the lines and it is difficult to pass the params?